### ISSUE TITLE
Support declaring the same predecessor more than once in SSABulder

### ISSUE THREAD

**Issue by greenhat** [0xPolygonMiden/miden-ir#38]
Extracted from https://github.com/0xPolygonMiden/miden-ir/pull/22#discussion_r1355858488

>A solution to this particular problem though is to split the edge in the predecessor, introducing new blocks along any duplicate edges which branch to the same successor, and which act as a passthrough for the "true" successor block. This gets you the desired behavior without the complexity of supporting the same block on more than one path in multi-way branching instructions.

**Comment by bitwalker** [comment]
@greenhat Do you know if this issue is still relevant?

**Comment by greenhat** [comment]
Yes. I have not addressed it yet, and the code is still the same.
